lib/srv/db/common/audit.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/defaults"
+	libevents "github.com/gravitational/teleport/lib/events"
+
+	"github.com/gravitational/trace"
+)
+
+// AuditConfig is the audit events emitter configuration.
+type AuditConfig struct {
+	// StreamWriter is used to emit audit events.
+	StreamWriter libevents.StreamWriter
+}
+
+// Check validates the config.
+func (c *AuditConfig) Check() error {
+	if c.StreamWriter == nil {
+		return trace.BadParameter("missing StreamWriter")
+	}
+	return nil
+}
+
+// Audit provides methods for emitting database access audit events.
+type Audit struct {
+	// cfg is the audit events emitter configuration.
+	cfg AuditConfig
+}
+
+// NewAudit returns a new instance of the audit events emitter.
+func NewAudit(config AuditConfig) (*Audit, error) {
+	if err := config.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &Audit{
+		cfg: config,
+	}, nil
+}
+
+// OnSessionStart emits an audit event when database session starts.
+func (a *Audit) OnSessionStart(ctx context.Context, session Session, err error) error {
+	event := &events.DatabaseSessionStart{
+		Metadata: events.Metadata{
+			Type: libevents.DatabaseSessionStartEvent,
+			Code: libevents.DatabaseSessionStartCode,
```

Comment:
I added the cluster name to the event metadata [not so long ago](https://github.com/gravitational/teleport/pull/4994/) - something to consider here and below as well?

Reply:
Good point, added.